0.preprocess-sites/2.process-cells.py

```diff
@@ -0,0 +1,178 @@
+
+        compartment_dir = pathlib.Path(batch_dir, site)
+
+        # Make the compartment_csvs dictionary used to merge dfs
+        compartment_csvs = {}
+        for compartment in compartments:
+            try:
+                metadata_cols = parent_cols[compartment.lower()] + id_cols
+            except KeyError:
+                metadata_cols = id_cols
+            compartment_csvs[compartment] = load_single_cell_compartment_csv(
+                compartment_dir, compartment, metadata_cols
+            )
+
+        # Load .tsv created in 1.process-spots
+        foci_file = pathlib.Path(
+            foci_dir, site, "full_cell_category_scores_by_guide.tsv.gz"
+        )
+        try:
+            foci_df = pd.read_csv(foci_file, sep="\t")
+        except FileNotFoundError:
+            print(
+                """Run 1.process-spots before continuing. \n
+                  If it has been run, check that output_basedir is set correctly in
+                  the process-spots section of the config"""
+            )
+
+        # Relabel columns in foci_df to start with Metadata_Foci_
+        foci_df.columns = [f"Metadata_Foci_{x}" if not x.startswith("Metadata_Foci") else x for x in foci_df.columns]
+
+    except FileNotFoundError:
+        print(f"{site} data not found")
+        continue
+
+    # Merge compartment csvs. Each row is a separate cell.
+    sc_merged_df = merge_single_cell_compartments(compartment_csvs, merge_info, id_cols)
+    sc_merged_df = sc_merged_df.sort_values(by=cell_sort_col)
+
+    # Add metadata to the merged csvs.
+    sc_merged_with_metadata_df = foci_df.merge(
+        sc_merged_df,
+        left_on=metadata_merge_foci_cols,
+        right_on=metadata_merge_cell_cols,
+        how="right",
+    )
+
+    # Drops all columns that are not Metadata (i.e. profiles)
+    metadata_df = sc_merged_with_metadata_df.loc[
+        :, sc_merged_with_metadata_df.columns.str.contains("Metadata")
+    ].drop_duplicates()
+
+    # Adds a cell quality category to previously uncategorized cells
+    metadata_df[metadata_cell_quality_col] = metadata_df[
+        metadata_cell_quality_col
+    ].fillna(empty_cell_category)
+
+    # Adds the site to the metadata_foci_site column to previously uncategorized cells
+    metadata_df[foci_site_col] = metadata_df[foci_site_col].fillna(site)
+
+    # Add the cell quality metadata to the df
+    metadata_df = (
+        metadata_df.merge(
+            cell_category_df,
+            left_on=metadata_cell_quality_col,
+            right_index=True,
+            how="left",
+        )
+        .sort_values(by=cell_sort_col)
+        .reset_index(drop=True)
+    )
+
+    # Create a summary of counts of each cell quality class
+    quality_counts = metadata_df.Cell_Class.value_counts()
+    cell_count_df = pd.DataFrame(quality_counts)
+    cell_count_df = cell_count_df.rename(columns={"Cell_Class": "cell_count"})
+    cell_count_df["site"] = site
```

Comment:
```suggestion
    cell_count_df = cell_count_df.assign(site=site)
```

lets switch to [`pd.assign()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.assign.html).